src/spec-node/dockerfileUtils.ts

```diff
@@ -138,7 +141,40 @@ function extractInstructions(stageStr: string) {
 		});
 }
 
+function handleExpressionVariables(dockerfile: Dockerfile, buildArgs: Record<string, string>, baseImageEnv: Record<string, string>, str: string, stage: { from?: From; instructions: Instruction[] }, beforeInstructionIndex: number) {
```

Comment:
Please add a few tests in dockerFileUtils.test.ts.

Reply:
Added tests.